src/MySqlConnector/MySqlClient/Results/ResultSet.cs

```diff
@@ -46,9 +47,39 @@ public ResultSet(MySqlDataReader dataReader)
 					if (State == ResultSetState.NoMoreData)
 						break;
 				}
-				else if (firstByte == 0xFB)
+				else if (firstByte == LocalInfilePayload.Signature)
 				{
-					throw new NotSupportedException("Don't support LOCAL_INFILE_Request");
+					var localInfile = LocalInfilePayload.Create(payload);
+					Stream infileStream;
+					try
+					{
+						if (localInfile.FileName.StartsWith(LocalInfilePayload.InfileStreamPrefix))
+						{
+							infileStream = MySqlBulkLoader.GetInfileStreamByKey(localInfile.FileName);
+						}
+						else
+						{
+							infileStream = new FileStream(localInfile.FileName, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, true);
```

Comment:
This can potentially throw lots of different exceptions, which will abort correct processing of the MySQL protocol.